### PR TITLE
Feature - Edit Template

### PR DESCRIPTION
**Pull request by Tandid**
- Created PATCH Request
- Changed Preview of Template to show only Exercise Name and Sets
- PATCH Request updated in Logging Client
- Clicking on Edit Icon opens Edit Template Modal
- User can add exercises, change number of sets, change name when editing template

### CODE REVIEW

File: src/app/api/user/logs/templates/[templateId]/route.ts [hunk 1]
```diff
@@ -1,62 +1,107 @@
...
+) {
+  if (!session?.user) {
+    return { status: 401, response: { error: "Unauthorized" } };
+  }
+
+  await dbConnect();
+  const user = await UserRepository.findById(session.user._id);
+
+  if (!user) {
+    return { status: 404, response: { error: "User not found" } };
```
Comment (davideastmond) [inline src/app/api/user/logs/templates/[templateId]/route.ts]: Let's return a `NextResponse.json` - just to keep our API responses consistent?

File: src/app/api/user/logs/templates/[templateId]/route.ts [hunk 2]
```diff
@@ -1,62 +1,107 @@
...
+  }
+
+  const templateIndex = user.logs.findIndex((log) => log._id == templateId);
+
+  if (templateIndex === -1) {
+    return { status: 404, response: { error: "Template not found" } };
```
Comment (davideastmond) [inline src/app/api/user/logs/templates/[templateId]/route.ts]: same here

File: src/app/api/user/logs/templates/[templateId]/route.ts [hunk 3]
```diff
@@ -1,62 +1,107 @@
...
+  }
+
+  return { user, templateIndex };
```
Comment (davideastmond) [inline src/app/api/user/logs/templates/[templateId]/route.ts]: same here

File: src/app/clients/logging-client/logging-client.ts [hunk 4]
```diff
@@ -74,4 +74,30 @@ export const LoggingClient = {
...
+      }
+    );
+
+    if (!response.ok) {
+      throw new Error(response.statusText);
+    }
+
+    const data = await response.json();
+
+    console.log("Template updated successfully!", data);
```
Comment (davideastmond) [inline src/app/clients/logging-client/logging-client.ts]: Do we need to commit this? Looks like test code?

File: src/app/user/mytemplates/page.tsx [hunk 5]
```diff
@@ -42,35 +51,55 @@ const MyTemplates: React.FC<MyTemplatesProps> = ({}) => {
...
+    if (template) {
+      setSelectedTemplate(template.exercises);
+      setSelectedTemplateName(template.name!);
+      setSelectedTemplateId(template._id!);
+      setIsEditModalOpen(true);
+    }
+  };
+
+  const handleUpdateTemplate = async (updatedTemplateData: Partial<Log>) => {
+    console.log("handleUpdateTemplate", updatedTemplateData);
```
Comment (davideastmond) [inline src/app/user/mytemplates/page.tsx]: Do we need to commit this console here?

File: src/components/modals/EditTemplateModal.tsx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+          setNumber: index + 1,
+          weight: 0,
+          unit: set.unit,
+          reps: 0,
+        })),
+      })),
+      isTemplate: true,
+    };
+
+    // console.log(updatedTemplateData);
```
Comment (davideastmond) [inline src/components/modals/EditTemplateModal.tsx]: Can we remove the commented out code?

File: src/components/modals/EditTemplateModal.tsx [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+    const newExercise: ExerciseActivity = {
+      exerciseName: exercise.label,
+      sets: [{ setNumber: 1, reps: 0, weight: 0, unit: "lbs" }],
+    };
+
+    setSelectedExercises((prev) => [...prev, newExercise]);
+    setSearchInput("");
+    setSearchResults([]);
+
+    return newExercise;
```
Comment (davideastmond) [inline src/components/modals/EditTemplateModal.tsx]: 🤔I noticed here an event handler was returning a value. Do we need to return a value for this function? Maybe I am missing something?

**Review by davideastmond — COMMENTED** [review]
Just some small things - otherwise looks good

File: src/components/modals/EditTemplateModal.tsx (hunk 7 above)
Comment (Tandid) [reply src/components/modals/EditTemplateModal.tsx]: Good catch! There should be no return value, the function is just supposed to update the state with the new exercise and reset the search.

**Review by davideastmond — APPROVED** [review]
👍🏿

**Pull request merged**